src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -281,6 +324,117 @@ public CallOutput CreateAccessList(BlockHeader header, Transaction tx, Cancellat
+                    }
+                }
+
+                foreach (MultiCallBlockStateCallsModel? callInputBlock in blocks)
+                {
+                    BlockHeader callHeader = null;
+                    if (callInputBlock.BlockOverride == null)
+                    {
+                        callHeader = new BlockHeader(
+                            parent.Hash,
+                            Keccak.OfAnEmptySequenceRlp,
+                            Address.Zero,
+                            UInt256.Zero,
+                            parent.Number + 1,
+                            parent.GasLimit,
+                            parent.Timestamp + 1,
+                            Array.Empty<byte>());
+                        callHeader.BaseFeePerGas = BaseFeeCalculator.Calculate(parent, _specProvider.GetSpec(parent));
+                    }
+                    else
+                    {
+                        callHeader = callInputBlock.BlockOverride.GetBlockHeader(parent, _blocksConfig);
+                    }
+
+                    env.StateProvider.StateRoot = parent.StateRoot;
+
+                    IReleaseSpec releaseSpec = _specProvider.GetSpec(parent);
+
+                    if (releaseSpec.IsEip4844Enabled)
+                    {
+                        // TODO: Calculate ExcessDataGas depending on parent ExcessDataGas and number of blobs in txs
+                        callHeader.ExcessDataGas = 0;
+                    }
+
+                    callHeader.MixHash = parent.MixHash;
+                    callHeader.IsPostMerge = parent.Difficulty == 0;
+
+                    var transactions = callInputBlock.Calls.Select(model => model.GetTransaction()).ToList();
+                    foreach (Transaction transaction in transactions)
+                    {
+                        transaction.SenderAddress ??= Address.SystemUser;
+
+                        Keccak stateRoot = callHeader.StateRoot!;
+
+                        if (transaction.Nonce == 0)
+                        {
+                            transaction.Nonce = env.StateProvider.GetAccount(transaction.SenderAddress).Nonce;
+                        }
+
+                        transaction.Hash = transaction.CalculateHash();
+                    }
+
+                    Block? currentBlock = new(callHeader, transactions, Array.Empty<BlockHeader>());
+
+                    var currentSpec = env.SpecProvider.GetSpec(currentBlock.Header);
+                    if (callInputBlock.StateOverrides != null)
+                    {
+                        ModifyAccounts(callInputBlock.StateOverrides, env.StateProvider, currentSpec);
+                    }
+
+                    env.StateProvider.Commit(currentSpec);
+                    env.StateProvider.CommitTree(currentBlock.Number);
+                    env.StateProvider.RecalculateStateRoot();
+
+                    currentBlock.Header.StateRoot = env.StateProvider.StateRoot;
```

Comment:
Let's move to helper method to override state

Reply:
moved to a local helper function